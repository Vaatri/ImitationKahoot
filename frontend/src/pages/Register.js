import React from 'react';
import { Link, useHistory } from 'react-router-dom';
import { Button, Container, TextField } from '@material-ui/core/';
import API from '../utils/api';

const api = new API('http://localhost:5005');

function Register() {
  function checkValidEmail(input) {
    return /^[\w-.]+@([\w-]+\.)+[\w-]{2,4}$/.test(input);
  }

  const [inputName, setName] = React.useState('');
  const [inputEmail, setEmail] = React.useState('');
  const [inputPassword, setPassword] = React.useState('');
  const [confirmPassword, setConfirmPassword] = React.useState('');
  const [validUser, setValidUser] = React.useState(false);
  const history = useHistory();

  const registerUser = async () => {
    // check if the shit user passed in is valid bruh
    if (inputPassword !== confirmPassword || !checkValidEmail(inputEmail)) {
      setValidUser(false);
    } else {
      setValidUser(true);
    }

    if (inputName && inputEmail && inputPassword && validUser
        && checkValidEmail(inputEmail)) {
      const options = {
        headers: { 'Content-type': 'application/json' },
        body: JSON.stringify({ email: inputEmail, password: inputPassword, name: inputName }),
      };
      const res = await api.post('admin/auth/register', options);
      if (res.token) {
        localStorage.setItem('user_token', res.token);
        history.push('/login');
      } else {
        // register is trash
      }
    }
  };

  return (
    <main>
      <Container>
        <h1>Register Cunts</h1>
        <form>
          <TextField label="Name*" variant="outlined" name="name" id="name" onChange={(event) => setName(event.target.value)} />
          <TextField label="Email*" variant="outlined" name="email" id="email" onChange={(event) => setEmail(event.target.value)} />
          <TextField label="Password" variant="outlined" name="password" type="password" id="password" onChange={(event) => setPassword(event.target.value)} />
          <TextField label="Confirm Password*" variant="outlined" name="confirm-password" type="password" id="confirm-password" onChange={(event) => setConfirmPassword(event.target.value)} />
          <Button onClick={() => registerUser()} variant="contained">Register</Button>
          <Link to="/login">If you already have an account click hereeeeeeeee</Link>
        </form>
      </Container>
    </main>
  );
}

export default Register;
