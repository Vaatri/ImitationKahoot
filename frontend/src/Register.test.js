import { shallow } from 'enzyme';
import React from 'react';
import Register from './pages/Register';

describe('Register', () => {
  const wrapper = shallow(<Register />);

  it('register title is "Register"', () => {
    const title = wrapper.find('#title');
    expect(title.text()).toEqual('Register');
  });

  it('takes a snapshot of register title', () => {
    const title = wrapper.find('#title');
    expect(title).toMatchSnapshot();
  });

  it('name label is "Name*', () => {
    const name = wrapper.find('#name');
    expect(name.props().label).toEqual('Name*');
  });

  it('takes a snapshot of name', () => {
    const name = wrapper.find('#name');
    expect(name).toMatchSnapshot();
  });

  it('email label is "Email*', () => {
    const email = wrapper.find('#email');
    expect(email.props().label).toEqual('Email*');
  });

  it('takes a snapshot of email', () => {
    const email = wrapper.find('#email');
    expect(email).toMatchSnapshot();
  });

  it('password label is "Password*', () => {
    const password = wrapper.find('#password');
    expect(password.props().label).toEqual('Password*');
  });

  it('takes a snapshot of password', () => {
    const password = wrapper.find('#password');
    expect(password).toMatchSnapshot();
  });

  it('submit button has aria-label', () => {
    const submit = wrapper.find('#submit');
    expect(submit.props()['aria-label']).toEqual('register');
  });

  it('submit button text should be "Register"', () => {
    const submit = wrapper.find('#submit');
    expect(submit.text()).toEqual('Register');
  });

  it('takes a snapshot of submit button', () => {
    const submit = wrapper.find('#submit');
    expect(submit).toMatchSnapshot();
  });

  it('simulate submit click', () => {
    const submit = wrapper.find('#submit');
    submit.simulate('click');
    const email = wrapper.find('#email');
    expect(email.props().error).toEqual(false);
  });

  it('redirect to login page button has aria-label', () => {
    const button = wrapper.find('#login');
    expect(button.props()['aria-label']).toEqual('login');
  });

  it('redirect to login page button text should be "If you already have an account click here to log in" ', () => {
    const button = wrapper.find('#login');
    expect(button.text()).toEqual('If you already have an account click here to log in');
  });

  it('takes snapshot of redirect button', () => {
    const button = wrapper.find('#login');
    expect(button).toMatchSnapshot();
  });
});
