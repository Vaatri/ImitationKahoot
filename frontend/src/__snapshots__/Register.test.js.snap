// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register takes a snapshot of email 1`] = `ShallowWrapper {}`;

exports[`Register takes a snapshot of name 1`] = `ShallowWrapper {}`;

exports[`Register takes a snapshot of password 1`] = `ShallowWrapper {}`;

exports[`Register takes a snapshot of register title 1`] = `ShallowWrapper {}`;

exports[`Register takes a snapshot of submit button 1`] = `ShallowWrapper {}`;

exports[`Register takes snapshot of redirect button 1`] = `ShallowWrapper {}`;
